### ISSUE TITLE
Is TypeAdapterConfig.GlobalSettings.ForDestinationType<T>().ConstructUsing(() => new MyType()) supported?

### ISSUE THREAD

**Issue by kellyethridge** [MapsterMapper/Mapster#169]
The configuration works, but if I call `GlobalSettings.Compile()` after defining the `ConstructUsing` configuration I get:
```
OneTimeSetUp: Mapster.CompileException : Error while compiling
source=System.Void
destination=MyType
type=Map
  ----> System.ArgumentException : Argument type cannot be System.Void.
```

**Comment by chaowlert** [comment]
Hi, thank you for bug report. I will try to inspect and fix soon.

**Comment by chaowlert** [comment]
fixed in 3.3.2
